<!-- Launches Gazebo with a basic Qubo robot spawned in the empty underwater world also spawns a thruster tranlator node. -->
<launch>
    <!-- Launch Gazebo with the empty underwater world -->
    <include file="$(find qubo_launch)/gazebo/empty_underwater_world.launch" />
    
    <!-- Spawn basic Qubo robot into the world -->
    <include file="$(find qubo_launch)/gazebo/spawn_basic_qubo.launch">
        <arg name="namespace" value="qubo_test"/>
    </include>

    <!-- spawn the thruster translator -->
    <node name="qubo_thruster_node" pkg="vl_qubo_gazebo" type="qubo_gazebo_hardware_node" />

    
      <!-- spawn the depth and yaw controllers -->
    <node name="qubo_depth_controller" pkg="controls" type="pid_controller" args="depth 100" >
      <param name="kp" value="0.0" />
      <param name="ki" value="0.0" />
      <param name="kd" value="0.0" />
      <param name="upper_limit" value="10.0" />
      <param name="lower_limit" value="-10.0" />
      <param name="windup_limit" value="5.0" />
      <param name="buffer_size" value ="10" />
    </node>
         
    
    <node name="qubo_surge_controller" pkg="controls" type="pid_controller" args="surge 100" >
      <param name="kp" value="0.0" />
      <param name="ki" value="0.0" />
      <param name="kd" value="0.0" />
      <param name="upper_limit" value="10.0" />
      <param name="lower_limit" value="-10.0" />
      <param name="windup_limit" value="10.0" />
    </node>
    
    <node name="qubo_sway_controller" pkg="controls" type="pid_controller" args="sway 100" >
      <param name="kp" value="0.0" />
        <param name="ki" value="0.0" />
        <param name="kd" value="0.0" />
        <param name="upper_limit" value="10.0" />
        <param name="lower_limit" value="-10.0" />
        <param name="windup_limit" value="10.0" />
    </node>
    
    
    <node name="qubo_roll_controller" pkg="controls" type="pid_controller" args="roll 100" >
      <param name="kp" value="0.0" />
      <param name="ki" value="0.0" />
      <param name="kd" value="0.0" />
      <param name="upper_limit" value="10.0" />
      <param name="lower_limit" value="-10.0" />
      <param name="windup_limit" value="10.0" />
      <param name="angular_variable" value="true"/>
    </node>
      
    <node name="qubo_pitch_controller" pkg="controls" type="pid_controller" args="pitch 100" >
      <param name="kp" value="0.0" />
      <param name="ki" value="0.0" />
      <param name="kd" value="0.0" />
      <param name="upper_limit" value="10.0.0" />
      <param name="lower_limit" value="-10.0.0" />
      <param name="windup_limit" value="10.0.0" />
      <param name="angular_variable" value="true" />
    </node>
      
    <node name="qubo_yaw_controller" pkg="controls" type="pid_controller" args="yaw 100" >
      <param name="kp" value="0.0" />
      <param name="ki" value="0.0" />
      <param name="kd" value="0.0" />
      <param name="upper_limit" value="10.0" />
      <param name="lower_limit" value="-10.0" />
      <param name="windup_limit" value="10.0" />
      <param name="angular_variable" value="true" />
    </node>

    
    <!-- spawn thing that let's us reconfigure parameters -->
    <node name="reconfigure_node" pkg="rqt_reconfigure" type="rqt_reconfigure" />

    <node name="plot_node" pkg="rqt_plot" type="rqt_plot" />
</launch>
